Converted all html file extensions and formatting in docs/resources folder from `html` to `md`

Dev-JoyA opened the pull request:
## Fixes
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'.  -->
<!-- If there is no issue being resolved, please open one before creating this pull request. -->
<!-- Replace [issue number] with the issue number (don't leave the square brackets)--likewise for [issue author]. -->
- Fixes #333  by @ Murdock9803

## Description
<!-- Concisely describe what the pull request does. -->
This PR addresses the issue of inconsistencies in the documentation resource files located in the docs/_resources directory. Currently, the  files in this folder are in two formats: Markdown (.md) and HTML (.html). This dual format leads to confusion among contributors and complicates the contribution process.




## Checklist
<!-- DON'T remove this section or any of the lines. -->
<!-- Leave incomplete or inapplicable lines unchecked. -->
<!-- Replace the [ ] with [x] to check the boxes (there is no space between x and square brackets). -->
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin
<!-- YOU MUST READ AND UNDERSTAND THE FOLLOWING ATTESTATION. -->
<!-- -->
<!-- Be aware that copying and pasting from discussion sites or generative AI isn't allowed under this DCO. -->

For the purposes of this DCO, "license" is equivalent to "license or public domain dedication," and "open source license" is equivalent to "open content license or public domain dedication."
<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

possumbilities commented on docs/Gemfile.lock:
```diff
@@ -270,6 +270,7 @@ GEM
 PLATFORMS
   aarch64-linux
   arm64-darwin-23
+  x86_64-linux
```
This is an extra change that needs to be removed from this PR.

possumbilities commented:
Hi @Dev-JoyA Thanks for doing this!

There are a number of locations where extra space is present preceding the first lines of paragraphs, and that would need to be corrected overall. 

Also, please verify that any of the html that is removed and replaced with markdown is not losing a class or changing its ultimate html structure.

Dev-JoyA commented:
> Hi @Dev-JoyA Thanks for doing this!
> 
> There are a number of locations where extra space is present preceding the first lines of paragraphs, and that would need to be corrected overall.
> 
> Also, please verify that any of the html that is removed and replaced with markdown is not losing a class or changing its ultimate html structure.

> I would correcting the issue with the spacing in another commit
> I ensured i replaced <ul> unordered list with its relevant markdown synthax, same with <a> and every other html structure are correctly replaced with markdown synthax.

possumbilities commented on docs/_resources/open-data-handbook.md:
```diff
@@ -16,4 +16,4 @@
 ---
 
-<p>A primer on the best practices for open data.</p>
+ A primer on the best practices for open data.
```
extra space

possumbilities commented on docs/_resources/how-to-attribute-creative-commons-photos.md:
```diff
@@ -16,4 +16,4 @@
 ---
 
-<p>Infographic explaining how to attribute CC-licensed photos.</p>
+ Infographic explaining how to attribute CC-licensed photos.
```
extra space

possumbilities commented on docs/_resources/big-buck-bunny-loves-creative-commons.md:
```diff
@@ -16,4 +16,5 @@
 ---
 
-<p>A fun video explaining how to use CC (<a href="">more information</a>).</p>
+A fun video explaining how to use CC ([more information](#))
```
It appears the link here is empty, we should either correct this, or perhaps make a new Issue documenting the missing hyperlink ref.

Dev-JoyA replied:
@possumbilities how can i get the correct link for this , so i can add it before pushing the next commit

possumbilities replied:
@Dev-JoyA You could try looking through the files history of changes and see if perhaps it was there in the past and was accidentally removed. If you can't find it it might just be worth documenting it as a separate Issue so others might track it down and correct it (since it is outside the scope of this Issue it just surface as as a result of this work)

Dev-JoyA replied:
@possumbilities i did a little research and i found this [BBB loves Creative Common](https://resources.creativecommons.org/big-buck-bunny-loves-creative-commons/).
Kindly confirm if this works so i can add it to the .md file

possumbilities replied:
I think that's it.

Dev-JoyA replied:
@possumbilities I have added it and pushed another commit

possumbilities reviewed (approved):
Approved! Thank you.

possumbilities commented:
@Dev-JoyA if you reply to #333 I can assign you.
